theme.js

```diff
@@ -0,0 +1,42 @@
+//Add transition class DOM and remove after 500ms
+let trans = () => {
+    document.documentElement.classList.add('transition');
+        window.setTimeout(() => {
+            document.documentElement.classList.remove('transition')
+        }, 500)
+}
+
+var isClicked;
+
+//Detect if browser prefers dark or light mode
+//Set isClicked according to preference
+
+const isDarkTheme = window.matchMedia("(prefers-color-scheme: dark)")
+if (isDarkTheme.matches) {
+        document.documentElement.setAttribute('data-theme', 'dark')
+        document.getElementById("icon").src="icon_dark.png";
+        isClicked = 1;
```

Comment:
For a toggle variable like this, prefer to use `true` and `false` instead of 1 and 0.